python_files/generate_figure_2.py

```diff
@@ -330,11 +330,53 @@
 plt.close()
 
 # cell types
-cell_classes = {'Immune': ['Immune', 'Mono_Mac', 'T', 'Granulocyte'], 'Cancer': ['Cancer'],
-                'Structural': ['Structural'], 'ECM': ['ecm'], 'Multiple': ['multiple', 'all', 'kmeans_cluster']}
-
-for cell_class in cell_classes.keys():
-    feature_metadata.loc[feature_metadata.cell_pop.isin(cell_classes[cell_class]), 'cell_class'] = cell_class
+feature_metadata = pd.read_csv(os.path.join(base_dir, 'analysis_files/feature_metadata.csv'))
+feature_metadata['cell_class'] = np.nan
+immune_cell_list = ['Mono_Mac'] + ['CD68_Mac', 'CD163_Mac', 'Mac_Other', 'Monocyte', 'APC'] + ['B'] + ['T'] + \
+                   ['CD4T', 'CD8T', 'Treg', 'T_Other'] + ['Granulocyte'] + ['Neutrophil', 'Mast'] + ['NK']
+cancer_cell_list = ['Cancer'] + ['Cancer_1', 'Cancer_2', 'Cancer_3']
+structural_cell_list = ['Structural'] + ['Endothelium', 'CAF', 'Fibroblast', 'Smooth_Muscle']
+
+for immune_cell in immune_cell_list:
+    feature_metadata.loc[np.logical_and(feature_metadata.feature_type == 'density', feature_metadata.feature_name.str.contains(immune_cell)), 'cell_class'] = 'Immune'
```

Comment:
Can we do something like `feature_metadata.feature_type.isin(['density', 'density_proportion', ... etc])`